modules/sor/sor.service.ts

```diff
@@ -0,0 +1,133 @@
+import { networkContext } from '../network/network-context.service';
+import { GqlCowSwapApiResponse, GqlSorSwapType, GqlSorGetSwapsResponse, GqlSorSwapOptionsInput } from '../../schema';
+import { sorV1BalancerService } from './sorV1Balancer/sorV1Balancer.service';
+import { sorV1BeetsService } from './sorV1Beets/sorV1Beets.service';
+import { sorV2Service } from './sorV2/sorV2.service';
+import { GetSwapsInput, SwapResult } from './types';
+import { EMPTY_COWSWAP_RESPONSE } from './constants';
+
+export class SorService {
+    public async getCowSwaps({
+        tokenIn,
+        tokenOut,
+        swapType,
+        swapAmount,
+    }: GetSwapsInput): Promise<GqlCowSwapApiResponse> {
+        console.time(`sorV1-${networkContext.chain}`);
+        const swapV1 = await sorV1BalancerService.getSwapResult({
+            tokenIn,
+            tokenOut,
+            swapType,
+            swapAmount,
+        });
+        console.timeEnd(`sorV1-${networkContext.chain}`);
+        console.time(`sorV2-${networkContext.chain}`);
+        const swapV2 = await sorV2Service.getSwapResult({
+            tokenIn,
+            tokenOut,
+            swapType,
+            swapAmount,
+        });
+        console.timeEnd(`sorV2-${networkContext.chain}`);
+
+        if (!swapV1.isValid && !swapV2.isValid) return EMPTY_COWSWAP_RESPONSE(tokenIn, tokenOut, swapAmount);
+
+        const bestSwap = this.getBestSwap(swapV1, swapV2, swapType, tokenIn, tokenOut);
+
+        try {
+            // Updates with latest onchain data before returning
+            return await bestSwap.getCowSwapResponse(true);
+        } catch (err) {
+            console.log(`Error Retrieving QuerySwap`);
+            console.log(err);
+            return EMPTY_COWSWAP_RESPONSE(tokenIn, tokenOut, swapAmount);
+        }
+    }
+
```

Comment:
does this also need something for the balancer UI (in the future) like `getBalancerSwaps()`?

Reply:
Not that I'm aware of.